funnel/models/session.py

```diff
@@ -23,15 +24,42 @@ class Session(UuidMixin, BaseScopedIdNameMixin, db.Model):
     proposal = db.relationship(Proposal,
         backref=db.backref('session', uselist=False, cascade='all, delete-orphan'))
     speaker = db.Column(db.Unicode(200), default=None, nullable=True)
-    start = db.Column(db.DateTime, nullable=False)
-    end = db.Column(db.DateTime, nullable=False)
+    start = db.Column(db.DateTime, nullable=True)
+    end = db.Column(db.DateTime, nullable=True)
     venue_room_id = db.Column(None, db.ForeignKey('venue_room.id'), nullable=True)
     venue_room = db.relationship(VenueRoom, backref=db.backref('sessions'))
     is_break = db.Column(db.Boolean, default=False, nullable=False)
     featured = db.Column(db.Boolean, default=False, nullable=False)
     banner_image_url = db.Column(db.Unicode(2000), nullable=True)
 
-    __table_args__ = (db.UniqueConstraint('project_id', 'url_id'),)
+    __table_args__ = (
+        db.UniqueConstraint('project_id', 'url_id'),
+        db.CheckConstraint(
+            u'("start" IS NULL AND "end" IS NULL) OR ("start" IS NOT NULL AND "end" IS NOT NULL)',
+            u'ck_session_start_end_nullable')
+        )
+
+    @hybrid_property
+    def scheduled(self):
+        # A session is scheduled only when both start and end fields have value
+        return (self.start != None) & (self.end != None)
```

Comment:
You'll need separate implementations for Python and SQLAlchemy. This should be the SQL version defined with `@scheduled.expression`